libmcount/dynamic.c

```diff
@@ -511,8 +511,63 @@ static bool skip_sym(struct sym *sym, struct mcount_dynamic_info *mdi,
 	return skip;
 }
 
-static void patch_func_matched(struct mcount_dynamic_info *mdi,
-			       struct uftrace_mmap *map)
+static void mcount_patch_func_with_stats(struct mcount_dynamic_info *mdi,
+					 struct sym *sym)
+{
+	switch (mcount_patch_func(mdi, sym, &disasm, min_size)) {
+	case INSTRUMENT_FAILED:
+		stats.failed++;
+		break;
+	case INSTRUMENT_SKIPPED:
+		stats.skipped++;
+		break;
+	case INSTRUMENT_SUCCESS:
+	default:
+		break;
+	}
+	stats.total++;
+}
+
+static void patch_patchable_func_matched(struct mcount_dynamic_info *mdi,
+					 struct uftrace_mmap *map)
+{
+	struct symtab *symtab;
+	unsigned long *patchable_loc = mdi->patch_target;
+	unsigned i;
+	struct sym *sym;
+	char namebuf[BUFSIZ];
+	struct sym fake_sym = {
+		.size = UINT_MAX,
```

Comment:
`.name = namebuf,`

Then you don't need to do it in the loop.

Reply:
ack